src/components/TodoItem.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todo: Todo
+  deleteTodo: (id: number) => void
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  deleteTodo = () => {},
+}) => {
+  const [isEdit, setIsEdit] = useState(false);
+
+  const handleDoubleClick = () => {
+    setIsEdit(true);
+  };
+
+  return (
+    <div className={classNames('todo', {
+      completed: todo.completed,
+    })}
+    >
+      <label className="todo__status-label">
+        <input
+          type="checkbox"
+          className="todo__status"
+          checked={todo.completed}
+        />
+      </label>
+
+      {isEdit ? (
+        <form>
+          <input
+            type="text"
+            className="todo__title-field"
+            placeholder="Empty todo will be deleted"
+            value="Todo is being edited now"
+          />
+        </form>
+      ) : (
+        <>
+          <span
+            className="todo__title"
+            onDoubleClick={handleDoubleClick}
+          >
+            {todo.title}
+          </span>
+
+          <button
+            type="button"
+            className="todo__remove"
+            onClick={() => deleteTodo(todo.id || 0)}
+          >
+            ×
+          </button>
+        </>
+      )}
+
+      {/* overlay will cover the todo while it is being updated */}
+      {/* 'is-active' class puts this modal on top of the todo */}
+      {/* <div className="modal overlay is-active"> */}
```

Comment:
Remove comment. Try it write obvious code that explains all we need without comments)